src/main/drivers/at32/serial_uart_at32bsp.c

```diff
@@ -182,6 +184,40 @@ void uartReconfigure(uartPort_t *uartPort)
     usart_enable(uartPort->USARTx,TRUE);
 }
 
+bool checkUsartTxOutput(uartPort_t *s)
+{
+    uartDevice_t *uart = container_of(s, uartDevice_t, port);
+    IO_t txIO = IOGetByTag(uart->tx.pin);
+
+    if ((uart->txPinState == TX_PIN_MONITOR) && txIO) {
+        if (IORead(txIO)) {
+            // TX is high so we're good to transmit
+
+            // Enable USART TX output
+            uart->txPinState = TX_PIN_ACTIVE;
+            IOConfigGPIOAF(txIO, IOCFG_AF_PP, uart->hardware->af);
+            return true;
+        } else {
+            // TX line is pulled low so don't enable USART TX
+            debug[s->port.identifier] = 2;
+            return false;
+        }
+    }
+
+    debug[s->port.identifier] = 1;
```

Comment:
(should be removed/changed to debug macro after testing)

Reply:
Debug now removed.